### DISCUSSION TITLE
Why does my template break?

### DISCUSSION THREAD

**Discussion by jfmercer** [twpayne/chezmoi#2208] [Q&A] [answered]
My goal is to have a script execute only if the OS is Darwin and `installMacApps` is set to `true`. But it breaks every time. 

My `.chezmoi.yaml.tmpl` looks like this:

```
{{- $interactive := stdinIsATTY -}}
{{- $email := "bob@gmail.com" -}}
{{- $installMacApps := true -}}

{{- if $interactive -}}
{{-   $email := promptString "email for .gitconfig" -}}
{{-   $installMacApps := eq (promptString "Install Mac applications (y/n)") "y" -}}
{{- end -}}

data:
  email: {{ $email | quote }}
  github_username: {{ $github_username | quote }}
  installMacApps: {{ $installMacApps }}
```

When I test the template, it looks good. `chezmoi execute-template --init < .chezmoi.yaml.tmpl` gives the following:

```
data:
  email: "bob@gmail.com"
  installMacApps: true
```

Yet, `chezmoi init && chezmoi apply` errors out: `chezmoi: template: .chezmoiscripts/darwin/run_onchange_after_configure_10_homebrew.sh.tmpl:1:37: executing ".chezmoiscripts/darwin/run_onchange_after_configure_10_homebrew.sh.tmpl" at <.installMacApps>: map has no entry for key "installMacApps"`.

That file begins with `{{- if and (eq .chezmoi.os "darwin") (eq .installMacApps true) -}}`. I also tried `{{- if and (eq .chezmoi.os "darwin") .installMacApps -}}`, but no luck. 

What I am doing wrong?

**Comment by twpayne** [comment, marked as answer]
> That file begins with {{- if and (eq .chezmoi.os "darwin") (eq .installMacApps true) -}}. I also tried {{- if and (eq .chezmoi.os "darwin") .installMacApps -}}, but no luck.

The problem here is that `.installMacApps` in the first line of the file is not defined when you run `chezmoi init` for the first time.

Note also this line:
    {{-   $installMacApps := eq (promptString "Install Mac applications (y/n)") "y" -}}
is incorrect. The `:=` creates a new `$installMacApps` variable that is scoped within the `if` statement. You need:
    {{-   $installMacApps = eq (promptString "Install Mac applications (y/n)") "y" -}}
instead. Note the `:=` is replaced with `=`: this modifies the existing `$installMacApps` variable.

**Reply by jfmercer** [reply]
@twpayne Thank you for your kind response. I had no idea that I was scoping those variables within the `if` statement. Perhaps I should study a little Go if I continue to use chezmoi extensively. In any case, you got me unstuck. I discovered that `chezmoi init` was not writing over config file, so no amount of reinitialization could get `installMacApps` defined. `chezmoi purge` fixed this for me. In any case, your comment got me unstuck & moving forward again. Many thanks!
